src/intersections.jl

```diff
@@ -73,6 +73,15 @@ struct CornerTouchingBoxes{P<:Point} <: Intersection
   value::P
 end
 
+# -------------------------------
+# SEGMENT-TRIANGLE INTERSECTIONS
+# -------------------------------
+
+struct IntersectingSegmentTri{P<:Point, T} <: Intersection
+  value::P
+  t::T
```

Comment:
What is this parameter t? Reading this struct I am assuming that IntersectingSegmentTri refers to the point case?